cmd/juju/application/upgradecharm.go

```diff
@@ -220,9 +218,6 @@ func (c *upgradeCharmCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *upgradeCharmCommand) Init(args []string) error {
 	switch len(args) {
 	case 1:
-		if !names.IsValidApplication(args[0]) {
```

Comment:
Can you please move this to a separate PR? It is not required for bug at hand. Then we can debate its merits there.

Most (I hope?) APIs are now using tags, and we do not currently have any tags that differ across API versions AFAIK (except Juju 1 & 2, of course). If we did, that would be an API break, and you would need to bump the facade version. The v2 controller continues to accept v1 tags, while also accepting v2 tags from clients who speak use the new facade version.